client/src/interface.rs

```diff
@@ -822,19 +823,18 @@ impl Stronghold {
         }
     }
 
-    /// Executes a runtime command at a remote Stronghold.
-    /// It is required that the peer has successfully been added with the `add_peer` method.
-    pub async fn remote_runtime_exec(&self, peer: PeerId, control_request: Procedure) -> ResultMessage<ProcResult> {
+    pub async fn remote_runtime_exec<P>(&self, peer: PeerId, control_request: P) -> ResultMessage<CollectedOutput>
+    where
+        P: Into<Procedure>,
+    {
         let actor = unwrap_or_err!(Option, self.network_actor, "No network actor spawned.");
         let send_request = network_msg::SendRequest {
             peer,
-            request: CallProcedure { proc: control_request },
+            request: control_request.into(),
         };
-        let receive_response = unwrap_or_err!(actor.send(send_request).await);
-        let result = unwrap_or_err!(receive_response);
-        match result {
-            Ok(ok) => ResultMessage::Ok(ok),
-            Err(err) => ResultMessage::Error(err.to_string()),
+        match unwrap_result_msg!(actor.send(send_request).await) {
```

Comment:
Good idea with macro, but what happens when a [MailboxError](https://docs.rs/actix/0.12.0/actix/enum.MailboxError.html) occurs?

Reply:
It would `return anyhow::anyhow!(mailbox_error)`. But I'll add a proper error type for this, instead of using anyhow.

Reply:
I concur.  A real error type would be useful here.